src/Dumpify/Renderers/Spectre.Console/TableRenderer/CustomTypeRenderers/DictionaryTypeRenderer.cs

```diff
@@ -125,24 +125,25 @@ private IEnumerable<(object? key, object? value)> GetPairs(IDescriptor descripto
                 if (genericArgument.GetGenericTypeDefinition() == typeof(KeyValuePair<,>))
                 {
                     var method = i.GetMethod("GetEnumerator", BindingFlags.Instance | BindingFlags.Public)!;
-                    var items = (IEnumerable)method.Invoke(obj, null)!;
-
+                    var getEnumeratorResult = method.Invoke(obj, null)!;
                     var list = new List<(object? key, object? value)>();
-
-                    foreach (var item in items)
+                    if (getEnumeratorResult is IEnumerator enumerator)
```

Comment:
I think that this is redundant. Since we call `i.GetMethod("GetEnumerator", BindingFlags.Instance | BindingFlags.Public)!;`, `items` must be an `IEnumerator` so we can change `(IEnumerable)method.Invoke(obj, null)!;` to `(IEnumerator)method.Invoke(obj, null)!;` and remove `if (getEnumeratorResult is IEnumerator enumerator)`.

Reply:
I will do the changes. I'll merge this PR :)